packages/portfolio-contract/src/portfolio.flows.ts

```diff
@@ -124,284 +112,302 @@ const initNobleAccount = async (
-  seat: ZCFSeat,
-  evmChain: SupportedEVMChains,
-  ctx: PortfolioContext,
-  protocol: YieldProtocol,
-  localAccount: LocalAccount,
 ) => {
-  const axelar = await orch.getChain('axelar');
-  const { chainId, stakingTokens } = await axelar.getChainInfo();
-  assert.equal(stakingTokens.length, 1, 'axelar has 1 staking token');
-
-  const caipChainId: CaipChainId = PositionChain[evmChain];
-  switch (protocol) {
-    case 'Aave':
-      keeper.initAave(caipChainId);
-      break;
-    case 'Compound':
-      keeper.initCompound();
-      break;
-    default:
-      assert.fail(protocol);
-  }
+  const { axelarChainsMap, contractAddresses, inertSubscriber } = ctx;
+  const { destinationEVMChain, gasAmount } = gmpArgs;
 
-  const { give } = seat.getProposal();
-
-  await ctx.zoeTools.localTransfer(seat, localAccount, { GMPFee: give.GMPFee });
-  try {
-    const memo: AxelarGmpOutgoingMemo = {
-      destination_chain: evmChain,
-      destination_address: ctx.contract.factory,
-      payload: [],
+  await sendGmp(
+    orch,
+    seat,
+    ctx,
+    harden({
+      destinationAddress: contractAddresses.factory,
+      destinationEVMChain,
       type: AxelarGMPMessageType.ContractCall,
-      fee: {
-        amount: '1', // TODO: Get fee amount from api
-        recipient: gmpAddresses.AXELAR_GAS,
-      },
-    };
+      gasAmount,
+      contractInvocationData: [],
+    }),
+    keeper,
+  );
 
-    trace('Calling factory contract via Axelar', memo);
-    const { AXELAR_GMP } = gmpAddresses;
-    await localAccount.transfer(
-      { value: AXELAR_GMP, encoding: 'bech32', chainId },
-      give.GMPFee,
-      { memo: JSON.stringify(memo) },
-    );
+  const addr = await keeper.getRemoteAddress();
+  const caipChainId = axelarChainsMap[destinationEVMChain].caip;
+  const accountId: AccountId = `${caipChainId}:${addr}`;
 
-    const getAddr = {
-      Aave: () => keeper.getAaveAddress(),
-      Compound: () => keeper.getCompoundAddress(),
-    }[protocol];
-    if (!getAddr) throw makeError(protocol);
-    const addr = await getAddr();
-    const accountId: AccountId = `${caipChainId}:${addr}`;
-
-    const topic: GuestInterface<ResolvedPublicTopic<unknown>> = {
-      description: `${protocol} EVM Addr`,
-      subscriber: ctx.inertSubscriber,
-      storagePath: accountId,
-    };
-    return topic;
-  } catch (err) {
-    console.error('⚠️ EVM account creation failed', err);
-    await ctx.zoeTools.withdrawToSeat(localAccount, seat, give);
-    assert.fail(`EVM account creation failed ${err}`);
-  }
+  const topic: GuestInterface<ResolvedPublicTopic<unknown>> = {
+    description: `EVM Addr`,
+    subscriber: inertSubscriber,
+    storagePath: accountId,
+  };
+  return topic;
 };
 
 const sendTokensViaCCTP = async (
-  orch: Orchestrator,
-  ctx: Pick<PortfolioContext, 'zoeTools' | 'chainHubTools'>,
+  orch,
+  seat,
+  ctx: PortfolioInstanceContext,
+  args: {
+    amount: Amount<'nat'>;
+    destinationEVMChain: AxelarChain;
+  },
```

Comment:
I can't see how this is supposed to be different; why not...

```suggestion
  args: BaseGmpArgs,
```

Reply:
It's the `key` difference. `BaseGmpArgs` contains `gasAmount` not `amount`. 
But you're right. I've made the update in `BaseGmpArgs` and all the places where it's used.